src/dotty/tools/dotc/core/pickling/PickleFormat.scala

```diff
@@ -0,0 +1,453 @@
+  final val ARTIFACT = 23
+  final val MUTABLE = 24
+  final val LABEL = 25
+  final val FIELDaccessor = 26
+  final val PARAMaccessor = 27
+  final val CASEaccessor = 28
+  final val COVARIANT = 29
+  final val CONTRAVARIANT = 30
+  final val SCALA2X = 31
+  final val DEFAULTparameterized = 32
+  final val DEFAULTinit = 33
+
+  final val SHARED = 96
+  final val TERMREFdirect = 97
+  final val TYPEREFdirect = 98
+  final val TERMREFstatic = 99
+  final val TYPEREFstatic = 100
+  final val BYTEconst = 101
+  final val BYTEneg = 102
+  final val SHORTconst = 103
+  final val SHORTneg = 104
+  final val CHARconst = 105
+  final val INTconst = 106
+  final val INTneg = 107
+  final val LONGconst = 108
+  final val LONGneg = 109
+  final val FLOATconst = 110
+  final val DOUBLEconst = 111
+  final val STRINGconst = 112
+
+  final val IDENT = 128
+  final val SELECT = 129
+  final val TERMREFsymbol = 130
+  final val TERMREF = 131
+  final val TYPEREFsymbol = 132
+  final val TYPEREF = 133
+
+  final val PACKAGE = 160
+  final val VALDEF = 161
+  final val DEFDEF = 162
+  final val TYPEDEF = 163
+  final val IMPORT = 164
+  final val TYPEPARAM = 165
+  final val PARAMS = 166
+  final val PARAM = 167
+  final val IMPORTED = 168
+  final val RENAMED = 169
+  final val APPLY = 170
+  final val TYPEAPPLY = 171
+  final val NEW = 172
+  final val PAIR = 173
+  final val TYPED = 174
+  final val NAMEDARG = 175
+  final val ASSIGN = 176
+  final val BLOCK = 177
+  final val IF = 178
+  final val CLOSURE = 179
+  final val MATCH = 180
+  final val RETURN = 181
+  final val TRY = 182
+  final val THROW = 183
+  final val SEQLITERAL = 184
+  final val JSEQLITERAL = 185
+  final val BIND = 186
+  final val ALTERNATIVE = 187
+  final val UNAPPLY = 188
+  final val ANNOTATED = 189
+  final val CASEDEF = 190
+  final val IMPLICITarg = 191
+  final val TEMPLATE = 192
+  final val THIS = 193
+  final val SUPER = 194
+  final val CLASSconst = 195
+  final val ENUMconst = 196
+  final val SUPERtype = 197
+  final val SKOLEMtype = 198
+  final val REFINEDtype = 199
+  final val APPLIEDtype = 200
+  final val TYPEBOUNDS = 201
+  final val TYPEALIAS = 202
+  final val ANDtype = 203
+  final val ORtype = 204
+  final val BYNAMEtype = 205
+  final val METHODtype = 206
+  final val POLYtype = 207
+  final val PARAMtype = 208
+  final val IMPLICITARG = 209
+  final val PRIVATEqualified = 210
+  final val PROTECTEDqualified = 211
+
+  final val firstSimpleTreeTag = EMPTYTREE
+  final val firstNatTreeTag = SHARED
+  final val firstTreeNatTreeTag = IDENT
+  final val firstLengthTreeTag = PACKAGE
+
+  def nameTagToString(tag: Int): String = tag match {
+    case UTF8 => "UTF8"
+    case QUALIFIED => "QUALIFIED"
+    case SIGNED => "SIGNED"
+    case EXPANDED => "EXPANDED"
+    case MODULECLASS => "MODULECLASS"
+    case SUPERACCESSOR => "SUPERACCESSOR"
+    case DEFAULTGETTER => "DEFAULTGETTER"
+  }
+
+  def astTagToString(tag: Int): String = tag match {
```

Comment:
This method is repetitive and could lead to divergence in printing and real names of nodes. Also, it adds a significant number of LoC. Could we merge this with the final val definitions above. 

With macros we could put it all in one line and write the name only once. However, until we have macros we could define a case class that will hold a pair of (byte, name). I believe that JVM would optimize most of the overhead away.